gapis/api/vulkan/transform_command_disabler.go

```diff
@@ -476,6 +483,25 @@ func (disablerTransform *commandDisabler) observeAndWriteCommand(cmd api.Cmd) er
 	return disablerTransform.writeCommand(cmd)
 }
 
+func (disablerTransform *commandDisabler) sortAndMergeDisabledCmds(ctx context.Context) {
+	api.SortSubCmdIDs(disablerTransform.disabledCommands)
+
+	size := len(disablerTransform.disabledCommands)
+	ids := make([]api.SubCmdIdx, 0, size)
+	ids = append(ids, disablerTransform.disabledCommands[size-1])
+
+	// Remove the duplicates and clean the subcommands of deleted subcommands
+	for i := int64(size - 2); i >= 0; i-- {
+		if ids[len(ids)-1].Contains(disablerTransform.disabledCommands[i]) {
+			continue
+		}
+		ids = append(ids, disablerTransform.disabledCommands[i])
+	}
+
+	api.ReverseSubCmdIDs(ids)
```

Comment:
I don't see why you loop in reverse order and then reverse the whole `ids` list? I think you can achieve the same while looping one, in order:

```suggestion
	// Remove the duplicates and clean the subcommands of deleted subcommands
	for _, cmd := range disablerTransform.disabledCommands {
		if len(ids) == 0 || !cmd.Contains(ids[len(ids)-1]) {
			ids = append(ids, cmd)
		} else {
			// Here cmd does contain the last element of ids, hence
			// disabling cmd will effectively disable ids[len(ids)-1],
			// so we replace ids[len(ids)-1] by cmd
			ids[len(ids)-1] = cmd
		}
	}
```

What do you think?

Reply:
I think you have a bit missing information here. The intention is: 

For this input list:

`[3893,0,6,3,0,22], [3893,0,6,3],[3893,0,6,3,0,13], [3893,0,6,3,0,28], [3893,0,11,19], [3893,0,6,3,0,17]`

sort them into:

`[3893,0,6,3,0,13],  [3893,0,6,3,0,17], [3893,0,6,3,0,22], [3893,0,6,3,0,28], [3893,0,6,3], [3893,0,11,19],`

then later deduct the unnecessary ones:

`[3893,0,6,3], [3893,0,11,19],`

If you run your code by hand, you can see that your output is different.

The problem is: the code you suggest only checks if the last command is contained by the current one but in our case there are multiple command has been contained by one.

I think any further iteration after here to eliminate reversing will make the code more complicated rather than simple. Reversing will not have any performance impact as we expect a small set of input commands for the usual case and semantically simpler than any trick to remove duplicates or deducting the subcommands of a disabled command.

Reply:
OK, thanks for the clarification. We could avoid the list reversal by sorting appropriately, e.g. in a way that leads to:

`[3893,0,6,3], [3893,0,6,3,0,13], [3893,0,6,3,0,17], [3893,0,6,3,0,22], [3893,0,6,3,0,28], [3893,0,11,19]`

such that we can then cleanup with this code:
```
	for _, cmd := range disablerTransform.disabledCommands {
		if len(ids) == 0 || !ids[len(ids)-1].Contains() {
			ids = append(ids, cmd)
		}
	}
```

How hard would it be to get the sorting as above?

If you just want to keep your current solution, I think you need to comment what the sorting leads to, so the removal procedure makes sense, e.g.:

```
// sorting leads to container commands being after their containee commands, e.g.:
// [1, 0, 2, 3], [1, 0, 2, 4], [1, 0, 2], ...
```

Reply:
That would require the change all the code that uses the sort logic which I don't know how much is used and how much is necessary.

I think a comment would be enough